res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,839 @@
+	var numTicks = (value < 0x64) ? value: (value - 128);
+	var newVal = engine.getValue(group, "mix") + numTicks/64*2;
+	engine.setValue(group, "mix", Math.max(0, Math.min(1, newVal)));
+};
+
+/* Next Rate range calculation */
+MC7000.getNextRateRange = function(currRateRange) {
+    for (var i = 0; i < MC7000.posRateRanges.length; i++) {
+        if (MC7000.posRateRanges[i] > currRateRange) {
+            return MC7000.posRateRanges[i];
+        }
+    }
+    return MC7000.posRateRanges[0];
+};
+
+/* Previous Rate range calculation */
+MC7000.getPrevRateRange = function(currRateRange) {
+    for (var i = 0; i < MC7000.negRateRanges.length; i++) {
+        if (MC7000.negRateRanges[i] < currRateRange) {
+            return MC7000.negRateRanges[i];
+        }
+    }
+    return MC7000.negRateRanges[0];
+};
+        
+/* LEDs for VuMeter */
+// VuMeters only for Channel 1-4 / Master is on Hardware
+MC7000.VuMeter = function(value, group) {
+	var deckNumber = script.deckFromGroup(group),
+	    peak = 0x76, // where the red LED starts (clipping indicator)
+        level = value*value*value*value*0x69;
+
+    if (engine.getValue(group, "PeakIndicator")) {
+        var level = peak;
+    }
+    // now send the level meter signal to controller
+    midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, level);
+};
+
+/* LEDs around Jog wheel */
+MC7000.JogLed = function(value, group) {
+	var deckNumber = script.deckFromGroup(group);
+	// do nothing before track starts
+	if (value < 0) return;
+	// While "VINYL" is active show spinny LEDs
+	if (MC7000.isVinylMode[deckNumber]) { 
+		var trackDuration = engine.getValue(group, "duration"),
+ 	        position = value * trackDuration / 60 * MC7000.scratchParams.recordSpeed,
+ 	        activeLED = Math.round(position * 96) % 96;
+ 	// While "VINYL" is off show track position
+ 	} else {
+  	    var activeLED = value * 96;
+  	};
+  	// sending the position of active LED to the controller
+    midi.sendShortMsg(0x90 + deckNumber -1, 0x06, activeLED);
+};
+
+// initial HotCue LED when loading a track with already existing hotcues
+MC7000.HotCueLED = function(value, group) {
+    var deckNumber = script.deckFromGroup(group);
+    if (MC7000.PADModeCue[deckNumber]) {
+        for (var i = 1; i <= 8; i++) {
+            if (value === 1) {
+                if (engine.getValue(group, "hotcue_"+i+"_enabled") === 1) {
+                    midi.sendShortMsg(0x94 + deckNumber - 1, 0x14 + i - 1, MC7000.padColor.hotcueon);
+                } 
+            } else {
+                if (engine.getValue(group, "hotcue_"+i+"_enabled") === 0) {
+                    midi.sendShortMsg(0x94 + deckNumber - 1, 0x14 + i - 1, MC7000.padColor.hotcueoff);
+                }
+            };
+        };
+    };
+};
+
+/* CONTROLLER SHUTDOWN */
+MC7000.shutdown = function () {
+	
+// Need to switch off LEDs one by one,
+// otherwise the controller cannot handle the signal traffic
```

Comment:
Can you explain further? AFAIK the messages would get buffered and then just sent out slower.

Reply:
In the Forum I asked that because I had a one liner packed into a loop switching off all LEDs by one. Be answered that some Controllers could not handle the signal traffic so I broke it down in several peaces which works just fine now.